### ISSUE TITLE
integer catalog number allows decimal

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#2154]
select CAT_NUM from cataloged_item where CAT_NUM_INTEGER!=round(CAT_NUM_INTEGER);
...
46 rows selected.

All seem to be UCM:Herp. Need to change that collection to string catalog number format or ROUND these, and prevent future introduction.

**Comment by dustymc** [comment]
More info:

```

select 
  guid_prefix,
  CAT_NUM 
from 
  cataloged_item,
  collection
where 
  cataloged_item.collection_id=collection.collection_id and
  CATALOG_NUMBER_FORMAT='integer' and
 10    CAT_NUM_INTEGER!=round(CAT_NUM_INTEGER);


GUID_PREFIX
------------------------------------------------------------
CAT_NUM
------------------------------------------------------------------------------------------------------------------------
UCM:Bird
1520.1

UCM:Bird
1520.2

UCM:Bird
586.1

UCM:Bird
586.2

UCM:Herp
1623.1

UCM:Herp
1623.2

UCM:Herp
2301.1

UCM:Herp
2301.2

UCM:Herp
3200.1

UCM:Herp
2067.1

UCM:Herp
2067.2

UCM:Herp
1981.2

UCM:Herp
1981.1

UCM:Herp
3643.1

UCM:Herp
3690.1

UCM:Herp
1041.5

UCM:Herp
1041.1

UCM:Herp
1041.2

UCM:Herp
1041.3

UCM:Herp
1041.4

UCM:Herp
3658.1

UTEP:Inv
6002.5

UTEP:Inv
4214.1

UTEP:Inv
2956.1

UTEP:Inv
3583.5

UTEP:Inv
6949.2

UTEP:Inv
5938.5

UTEP:Inv
3582.5

UTEP:Inv
1295.5

UTEP:Inv
5426.1

UTEP:Inv
6949.1

UTEP:Inv
4530.1

UTEP:Inv
2468.1

UTEP:Inv
2031.1

UTEP:Inv
2832.1

UTEP:Inv
3602.1

UTEP:Inv
2453.1

UTEP:Inv
2517.1

UTEP:Inv
7767.1

UTEP:Inv
11956.1

UTEP:Inv
11956.2

UTEP:Inv
9285.1

UTEP:Inv
6983.1

UTEP:Herb
2725.1


44 rows selected.
```

This will need resolved at some point - these collections could be moved to a more relaxed catalog number format (which of course gives up some capabilities), or the catalog numbers will need to become integers. [Redirects](documentation) are usually a good idea if the catalog numbers change.

**Comment by Jegelewicz** [comment]
@mvzhuang

**Comment by AJLinn** [comment]
Could the decimals change to dashes? We have already make that fix work with the UAM:EH, UAM:Arc, and UAM:Art catalog numbers.

**Comment by dustymc** [comment]
For a non-integer catalog format, dashes, dots, whatever are fine. (And that comes with the loss of tools - eg, I can't know what's "next" after 1234.5 - 1234.5000000000000001, 1235, 1234.6, ????)

For collections with integer-format catalog numbers, we need to restrict the values to actual integers. 

(This is basically my fault - Oracle has a fairly generic NUMBER format and I didn't create a more-strict datatype for these. A tool I've been exploring has a strict INTEGER datatype, and the not-integer NUMBERS won't fit. Sorry!!)

**Comment by mvzhuang** [comment]
I might be being paranoid, but is there a problem with changing things to string format and then back to integer once we've resolved our decimal numbers? I don't have time to fix these at the moment.

**Comment by dustymc** [comment]
@mvzhuang that's what we'll do if this becomes pressing. I'd expect some minor problems in converting back to integer, but we can figure it out.

This is not critical for me at the moment. If something drastic happens in some /internal issues it might get that way, but I think it's very unlikely to become a real problem in the next few months. If you get to it before then great, if not we'll do what we have to when we have to.

**Comment by dustymc** [comment]
This just became a blocker, so unless I hear otherwise very soon I'll add 10M and round for "integer" catalog numbers which aren't.

@mvzhuang @ebraker @Jegelewicz 

```

 guid_prefix | cat_num 
-------------+---------
 UTEP:Herb   | 2725.1
 UTEP:Inv    | 6002.5
 UTEP:Inv    | 4214.1
 UTEP:Inv    | 2956.1
 UTEP:Inv    | 3583.5
 UTEP:Inv    | 6949.2
 UTEP:Inv    | 5938.5
 UTEP:Inv    | 3582.5
 UTEP:Inv    | 1295.5
 UTEP:Inv    | 5426.1
 UTEP:Inv    | 6949.1
 UTEP:Inv    | 4530.1
 UTEP:Inv    | 2468.1
 UTEP:Inv    | 2031.1
 UTEP:Inv    | 2832.1
 UTEP:Inv    | 3602.1
 UTEP:Inv    | 2453.1
 UTEP:Inv    | 2517.1
 UTEP:Inv    | 7767.1
 UTEP:Inv    | 11956.1
 UTEP:Inv    | 11956.2
 UTEP:Inv    | 9285.1
 UTEP:Inv    | 6983.1
 UCM:Bird    | 1520.1
 UCM:Bird    | 1520.2
 UCM:Bird    | 586.1
 UCM:Bird    | 586.2
 UCM:Herp    | 1623.1
 UCM:Herp    | 1623.2
 UCM:Herp    | 2301.1
 UCM:Herp    | 2301.2
 UCM:Herp    | 3200.1
 UCM:Herp    | 2067.1
 UCM:Herp    | 2067.2
 UCM:Herp    | 1981.2
 UCM:Herp    | 1981.1
 UCM:Herp    | 3643.1
 UCM:Herp    | 3690.1
 UCM:Herp    | 1041.5
 UCM:Herp    | 1041.1
 UCM:Herp    | 1041.2
 UCM:Herp    | 1041.3
 UCM:Herp    | 1041.4
 UCM:Herp    | 3658.1
```

**Comment by dustymc** [comment]
Of course it can't be as simple as adding.

New catalog numbers are "new_new_catnum" in the attached, I created "institutional catalog number" for the old and added redirects, the old URLs should continue to work.

[temp_newcatnums.csv.zip](https://github.com/ArctosDB/arctos/files/6273351/temp_newcatnums.csv.zip)

**Comment by itzuecs** [comment]
I am not sure how this fix the issue #3334 Request - sort by catalog number not by alphanumeric (GUID). The numbers in our catalogue are still not in order on Arctos. Do you mind explaining how this will help with the ordering to collections that do not need the new_new_catnum?
